src/coreclr/src/vm/methodtablebuilder.cpp

```diff
@@ -2365,6 +2444,9 @@ MethodTableBuilder::EnumerateMethodImpls()
                 {
                     BuildMethodTableThrowException(IDS_CLASSLOAD_MI_MISSING_SIG_BODY);
                 }
+
+                BOOL allowCovariantReturn = IsEligibleForCovariantReturns(theDecl);
```

Comment:
Would it make sense to do this only if the ordinary `CompareMethodSigs` fails?

Reply:
It's possible yes. I guess that actually makes more sense, since probably 90% of the cases won't have covariant return types, so we can skip eligibility checking here.